### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #12: Add "Contribution" section to readme
List some different ways how to contribute to LibrePCB, for example:
- coding: fix issues (starting with issues labeled with "easy")
- design/coding: create a website? design icons?
- donate: [Gratipay](https://gratipay.com/)?
- donate: PayPal?
- donate: Bitcoin?
- other ways to contribute?

also an IRC channel would be nice...

### PR DESCRIPTION
**Pull request by ubruhin**
fixes #12

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - To quickly create a topic branch based on master: 
+    `git checkout -b my_contribution master`
+  - Please avoid working directly on the `master` branch.
+- Write code which follows our [code style guides](http://librepcb.org/LibrePCB-Doxygen/master/df/d24/doc_code_style_guide.html)
+  and [.editorconfig settings](https://github.com/LibrePCB/LibrePCB/blob/master/.editorconfig).
+- Make commits of logical units.
+  - Make sure your commit messages are in the proper format.
+    Example: [0d896b69fdffbeb873957ed2b66eb9b0aa2ba750](https://github.com/LibrePCB/LibrePCB/commit/0d896b69fdffbeb873957ed2b66eb9b0aa2ba750)
+- Make sure you have added the necessary tests for your changes.
+- Run _all_ the tests to assure nothing else was accidentally broken.
```
Comment (The-Compiler) [inline CONTRIBUTING.md]: You might want to explain how to do that.

Comment (rnestler) [inline CONTRIBUTING.md]: Is it really *that* important to run the tests locally? I mean we have CI...

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Notes
+
+- **Before spending lots of time on something, ask for feedback on your idea first!**
+- Please search issues and pull requests before adding something new to avoid duplicating efforts and conversations.
+- Issues which are considered to be easy to solve are marked with the label `easy`: [Browse all easy issues](https://github.com/LibrePCB/LibrePCB/labels/easy).
+
+## Getting Started
+
+- Make sure you have a [GitHub account](https://github.com/signup/free).
```
Comment (rnestler) [inline CONTRIBUTING.md]: Can't we assume that? Who else is going to read the Contributing.md?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+- Make commits of logical units.
+  - Make sure your commit messages are in the proper format.
+    Example: [0d896b69fdffbeb873957ed2b66eb9b0aa2ba750](https://github.com/LibrePCB/LibrePCB/commit/0d896b69fdffbeb873957ed2b66eb9b0aa2ba750)
+- Make sure you have added the necessary tests for your changes.
+- Run _all_ the tests to assure nothing else was accidentally broken.
+
+## Submitting Changes
+
+- Push your changes to a topic branch in your fork of the repository.
+- Submit a pull request to the repository in the LibrePCB organization.
```
Comment (rnestler) [inline CONTRIBUTING.md]: Maybe "create a pull reqeust"? Submit seems strange to me.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dbrgn) [inline CONTRIBUTING.md]: It reduces the feedback loop for contributors. Waiting 30 minutes for CI sucks.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+- Open a new issue for your idea, assuming one does not already exist.
+- Fork the repository on GitHub.
+- Have a look at our [development resources](https://github.com/LibrePCB/LibrePCB/tree/master/dev), 
+  especially at the [Doxygen documentation](http://librepcb.org/LibrePCB-Doxygen/master/).
+- When using QtCreator, import and use our [code style guide file](https://github.com/LibrePCB/LibrePCB/blob/master/dev/CodingStyle_QtCreator.xml).
+
+## Making Changes
+
+- Create a topic branch from where you want to base your work.
+  - This is usually the master branch.
```
Comment (dbrgn) [inline CONTRIBUTING.md]: I'd remove this line. It adds no useful information, the next line already mentions master.

File: CONTRIBUTING.md (hunk 1 above)
Comment (The-Compiler) [inline CONTRIBUTING.md]: I agree, but it should be worded as an useful hint, not as a "command"/recommendation

File: CONTRIBUTING.md (hunk 2 above)
Comment (The-Compiler) [inline CONTRIBUTING.md]: Wait what, there are people without a GitHub account? :open_mouth:

File: CONTRIBUTING.md (hunk 3 above)
Comment (The-Compiler) [inline CONTRIBUTING.md]: Looks fine to me FWIW.

File: CONTRIBUTING.md (hunk 1 above)
Comment (dbrgn) [inline CONTRIBUTING.md]: By the way, "assure" should be "ensure".

File: CONTRIBUTING.md (hunk 2 above)
Comment (ubruhin) [inline CONTRIBUTING.md]: @rnestler One can read the CONTRIBUTING.md _without_ having a GitHub account ;)

File: CONTRIBUTING.md (hunk 3 above)
Comment (ubruhin) [inline CONTRIBUTING.md]: I would also say it's fine...
Btw, it's stolen from https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md ;)

File: CONTRIBUTING.md (hunk 4 above)
Comment (ubruhin) [inline CONTRIBUTING.md]: But the next line is only an example. So the recommendation to use the master branch _is_ an useful information in my opinion.

**Comment by ubruhin** [comment]
I've updated the pull request. #83 is now also considered. Any other comments/ideas?

**Pull request merged**